### PR TITLE
add timestamp-filters

### PR DESCRIPTION
**Pull request by bennsimon**

### CODE REVIEW

**Review by githengi — CHANGES_REQUESTED** [review]
Use OpenMRS formatter to format the code

File: src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java [hunk 1]
```diff
@@ -383,7 +383,11 @@ private ClientMetadata createMetadata(Client client, Long clientId) {
 			clientMetadata.setFirstName(client.getFirstName());
 			clientMetadata.setMiddleName(client.getMiddleName());
 			clientMetadata.setLastName(client.getLastName());
-			
+
+			if(clientId != null){
```
Comment (ndegwamartin) [inline src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:387]: Do we need to add the complementary branch of this if condition for created at?

File: src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java [hunk 2]
```diff
@@ -714,6 +773,10 @@ private LocationMetadata createMetadata(PhysicalLocation entity, Long id) {
 			locationMetadata.setVersion(entity.getProperties().getVersion());
 		}
 		locationMetadata.setServerVersion(entity.getServerVersion());
+
+		if(id != null){
```
Comment (ndegwamartin) [inline src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java:777]: Do we need to add the complementary branch of this if condition for created at?

File: src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java [hunk 3]
```diff
@@ -731,6 +794,10 @@ private StructureMetadata createStructureMetadata(PhysicalLocation entity, Long
 			}
 		}
 		structureMetadata.setServerVersion(entity.getServerVersion());
+
+		if(id != null){
+			structureMetadata.setDateEdited(new Date());
+		}
```
Comment (ndegwamartin) [inline src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java:800]: Do we need to add the complementary branch of this if condition for created at?

File: src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java (hunk 1 above)
Comment (bennsimon) [reply src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java:387]: Added it here https://github.com/OpenSRP/opensrp-server-core/blob/acbdf68075023cfb16a524f1ace88a6218c69b42/src/main/java/org/opensrp/repository/postgres/ClientsRepositoryImpl.java#L142

File: src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java (hunk 2 above)
Comment (bennsimon) [reply src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java:777]: https://github.com/OpenSRP/opensrp-server-core/blob/acbdf68075023cfb16a524f1ace88a6218c69b42/src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java#L171

File: src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java (hunk 3 above)
Comment (bennsimon) [reply src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java:800]: https://github.com/OpenSRP/opensrp-server-core/blob/acbdf68075023cfb16a524f1ace88a6218c69b42/src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java#L192

Comment (ndegwamartin) [reply src/main/java/org/opensrp/repository/postgres/LocationRepositoryImpl.java:800]: I think metadata never gets populated with date created in the _createMetadata_ method?

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**